tmail-backend/mailbox/team-mailboxes/src/test/scala/com/linagora/tmail/team/TeamMailboxTest.scala

```diff
@@ -0,0 +1,42 @@
+package com.linagora.tmail.team
+
+import com.linagora.tmail.team.TeamMailboxRepositoryContract.TEAM_MAILBOX_USERNAME
+import org.apache.james.mailbox.model.MailboxPath
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class TeamMailboxTest {
+
+  @Test
+  def fromMailboxPathShouldReturnNoneWhenNamespaceInvalid(): Unit = {
+    assertThat(TeamMailbox.from(new MailboxPath("namespace123", TEAM_MAILBOX_USERNAME, "sales"))
+      .isEmpty)
+      .isTrue
+  }
+
+  @Test
+  def fromMailboxPathShouldReturnNoneWhenNameInvalid(): Unit = {
+    assertThat(TeamMailbox.from(new MailboxPath("namespace123", TEAM_MAILBOX_USERNAME, "sales."))
+      .isEmpty)
+      .isTrue
+  }
+
+  @Test
+  def fromMailboxPathShouldReturnTeamMailboxWhenValid(): Unit = {
+    assertThat(TeamMailbox.from(new MailboxPath("#TeamMailbox", TEAM_MAILBOX_USERNAME, "sales"))
+      .isEmpty)
+      .isFalse
```

Comment:
isEqual to the team mailbox... you should check that we get the result we want, isFalse means it returns something but is it the correct result we want?